app/controllers/comments_controller.rb

```diff
@@ -11,14 +11,16 @@ def edit
   def create
     @comment = Comment.new(comment_params)
     @comment.user = current_user
-
     respond_to do |format|
       if @comment.save
+        reference = @comment.root.commentable
         format.html { redirect_to :back, notice: 'Comment was successfully created.' }
         format.json { render :show, status: :created, location: @comment }
+        format.js { render( @comment.parent.nil? ? 'comment_on_reference.js.erb' : 'reply_to_comment.js.erb', locals: {reference: reference} )}
```

Comment:
I feel like it should only require one jquery partial.

Also, you and your ternary operators. When these ternary operator lines get really long, I recommend breaking them out into `if/then/else` statements.

For this one in particular, you can do 

```ruby
format.js do
  if @comment.parent
    render 'reply_to_comment.js.erb', locals: {reference: reference}
  else
    render 'comment_on_reference.js.erb', locals: {reference: reference}
  end
end
```

Way more readable.

That said, I strongly feel like it shouldn't require two separate partials.